src/components/entity/ha-entity-multi-picker.ts

```diff
@@ -0,0 +1,330 @@
+import "@material/mwc-button/mwc-button";
+import "@polymer/paper-tooltip/paper-tooltip";
+import { mdiClose, mdiPlus } from "@mdi/js";
+import { HassEntity, UnsubscribeFunc } from "home-assistant-js-websocket";
+import { css, CSSResultGroup, html, LitElement, unsafeCSS } from "lit";
+import { customElement, property, query, state } from "lit/decorators";
+import { classMap } from "lit/directives/class-map";
+import { fireEvent } from "../../common/dom/fire_event";
+import { ensureArray } from "../../common/ensure-array";
+import { computeStateName } from "../../common/entity/compute_state_name";
+import {
+  EntityRegistryEntry,
+  subscribeEntityRegistry,
+} from "../../data/entity_registry";
+import { SubscribeMixin } from "../../mixins/subscribe-mixin";
+import { HomeAssistant } from "../../types";
+import type { HaDevicePickerDeviceFilterFunc } from "../device/ha-device-picker";
+import type { HaEntityPickerEntityFilterFunc } from "./ha-entity-picker";
+import "../ha-icon-button";
+import "../ha-state-icon";
+import "../ha-svg-icon";
+
+@customElement("ha-entity-multi-picker")
+export class HaEntityMultiPicker extends SubscribeMixin(LitElement) {
+  @property({ attribute: false }) public hass!: HomeAssistant;
+
+  @property({ type: Boolean }) public disabled?: boolean;
+
+  @property() public label?: string;
+
+  @property() public value?: any;
+
+  /**
+   * Show only entitys with entities from specific domains.
+   * @type {Array}
+   * @attr include-domains
+   */
+
+  @property({ type: Array, attribute: "include-domains" })
+  public includeDomains?: string[];
+
+  /**
+   * Show no entitys with entities of these domains.
+   * @type {Array}
+   * @attr exclude-domains
+   */
+  @property({ type: Array, attribute: "exclude-domains" })
+  public excludeDomains?: string[];
+
+  /**
+   * Show only entitys with entities of these device classes.
+   * @type {Array}
+   * @attr include-device-classes
+   */
+  @property({ type: Array, attribute: "include-device-classes" })
+  public includeDeviceClasses?: string[];
+
+  @property() public deviceFilter?: HaDevicePickerDeviceFilterFunc;
+
+  @property() public entityFilter?: HaEntityPickerEntityFilterFunc;
+
+  @property({ type: Boolean, attribute: "multiple" })
+  public multiple?: boolean;
+
+  @state() private _entities?: { [entityId: string]: EntityRegistryEntry };
+
+  @state() private _addMode?: "entity_id";
+
+  @query("#input") private _inputElement?;
+
+  public hassSubscribe(): UnsubscribeFunc[] {
+    return [
+      subscribeEntityRegistry(this.hass.connection!, (entities) => {
+        const entityLookup: { [entityId: string]: EntityRegistryEntry } = {};
+        for (const entity of entities) {
+          entityLookup[entity.entity_id] = entity;
+        }
+        this._entities = entityLookup;
+      }),
+    ];
+  }
+
+  protected render() {
+    if (!this._entities) {
+      return html``;
+    }
+    return html`
+      <div class="mdc-chip-set items">
```

Comment:
this code needs to be generalized. This is now copy pasted 3 times?  

I think it might benefit it we start with a single selector instead of trying to do entity, device and areas in one PR.

Reply:
How about a single selector file with a common base class and 4 specializations (Target, Area, Device, Entity) ?
Then we have the existing Area, Device and Entity pickers that are called from the selectors.